src/run.ps1

```diff
@@ -51,16 +69,94 @@ function setPathForAzCliCert($config)
+        return $null
+    }
+    $content = $response.Content
+    foreach ($line in $content -split "`n") {
+        if ($line.StartsWith('VERSION')) {
+            $match = [System.Text.RegularExpressions.Regex]::Match($line, 'VERSION = "(.*)"')
+            if ($match.Success) {
+                return $match.Groups[1].Value
+            }
+        }
+    }
+    logWarn "Failed to extract the latest version from the content of '$gitUrl'"
+    return $null
 }
 
-function deactivate_venv()
-{
-    Write-Host "De-ctivating python venv..."
-    deactivate
+function shouldInstallAzCli() {
+    # This function returns a boolean value, but any undirected / uncaptured stdout
+    # inside the function might be interpreted as true value by the caller.
+    # We can redirect using *>> to avoid this.
+    logH2 "Validating and installing 64-bit azure-cli"
+    $azCmd = (Get-Command az -ErrorAction SilentlyContinue)
+    if ($null -eq $azCmd) {
+        logText "Azure CLI is not installed. Installing..."
+        return $true
+    }
+
+    $currentAzVersion = az version --query '\"azure-cli\"' -o tsv 2>> $logFile
+    logText "Azure CLI version $currentAzVersion found in PATH at location: '$($azCmd.Source)'"
+    $azVersion = az --version *>&1;
+    $azVersionLines = $azVersion -split "`n"
+    
+    $pyLoc = $azVersionLines | Where-Object { $_ -match "^Python location" }
+    if ($null -eq $pyLoc) {
+        logWarn "Warning: Python location could not be found from the output of az --version:`n$($azVersionLines -join "`n"))"
+        return $true
+    }
+    logText $pyLoc
+    $pythonExe = $pyLoc -replace "^Python location '(.+?)'$", '$1'
+    try {
+        logText "Determining the bitness of Python at '$pythonExe'"
+        $arch = & $pythonExe -c "import struct; print(struct.calcsize('P') * 8)";
+        if ($arch -lt 64) {
+            logText "Azure CLI is $arch-bit. Installing 64-bit version..."
+            return $true
+        }
+    }
+    catch {
+        logText "Warning: Python version could not be determined from the output of az --version:`n$($azVersionLines -join "`n"))"
+        return $true
+    }
+
+    logH2 "$arch-bit Azure CLI is already installed. Checking for updates..."
+    $latestAzVersion = getLatestAzVersion
+    if ($latestAzVersion -and ($latestAzVersion -ne $currentAzVersion)) {
+        logText "A newer version of Azure CLI ($latestAzVersion) is available, installing it..."
+        return $true
+    }
+    logText "Azure CLI is up to date."
+    return $false
+}
+
+function installAzCli64Bit() {
+    $azCliMsi = "https://aka.ms/installazurecliwindowsx64"
+    $azCliMsiPath = Join-Path $PSScriptRoot "AzureCLI.msi"
+    $msiInstallLogPath = Join-Path $env:Temp "azInstall.log"
+    logText "Downloading Azure CLI 64-bit MSI from $azCliMsi to $azCliMsiPath"
+    Invoke-WebRequest -Uri $azCliMsi -OutFile $azCliMsiPath
```

Comment:
Do we not need try catch block here?

Reply:
This also has been leveraged from Connected-vmware team. But it is good to catch failures. Are you saying we need to throw exception of the download is not success?

Reply:
Right